frame/ddc-metrics-offchain-worker/src/lib.rs

```diff
@@ -297,8 +319,64 @@ impl<T: Trait> Module<T> {
         Ok(signer)
     }
 
+    fn sc_get_current_period_ms() -> ResultStr<u64> {
+        let contract_id = T::ContractId::get();
+        let contract_idx = <<T::CT as frame_system::Trait>::Lookup as StaticLookup>::lookup(contract_id).unwrap();
+
+        let call_data = Self::encode_get_current_period_ms();
+        let (exec_result, _gas_consumed) = pallet_contracts::Module::<T::CT>::bare_call(
+            Default::default(),
```

Comment:
What is default here?

Reply:
It’s the sender of the transaction, which does not matter because it is a getter.

Reply:
Default is used because of this method is called by any one without signature. @naure , is that right?

Reply:
Yes, it is not a transaction and the caller does not matter.

Reply:
The `default` method comes from the line `#[derive(Default)]` on structs. It returns empty or all-zeros values.